### PR TITLE
Adding NengoEdge client docs

### PR DESCRIPTION
**Pull request by bjkomer**

### CODE REVIEW

**Review by tbekolay — APPROVED** [review]
Made the changes discussed over the past week or so, and it's all looking decent to me now. Will merge after @drasmuss has a chance to look it over (or whoever).

File: docs/tutorials/uploading-datasets.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
+Use your own audio data in NengoEdge
```
Comment (drasmuss) [inline docs/tutorials/uploading-datasets.rst:1]: We should mark this specifically as "audio data for KWS" (and highlight that in the text as well), to make it clear that we're talking about the format for KWS datasets.

File: docs/tutorials/uploading-datasets.rst [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+Datasets in NengoEdge are split into
+three main folders named train, validation and test.
+Inside each of these main folders,
+labelled data is stored as ``.wav``, ``.flac`` or ``.mp3`` files
+organized into sub-folders,
+where the name of each sub-folder is the keyword label.
+Place all audio samples associated with that label
+into the corresponding sub-folder,
+then create a ``.zip`` or ``.tar.gz`` archive containing
```
Comment (drasmuss) [inline docs/tutorials/uploading-datasets.rst:22]: No .zip

File: docs/tutorials/intro-gpu.rst [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+so shorter test runs save money in the long run.
+You can reuse the weights from the short run as the starting point for longer runs,
+so there's no downside to a test run.*
+
+1. Click on **Runs** in the top bar.
+
+2. Click on **Create new project** to organize the runs for this project.
+   We will use the name "Directions"
+
+3. Under the Directions heading, click **New run**.
```
Comment (drasmuss) [inline docs/tutorials/intro-gpu.rst:71]: The button is "+KWS" now

File: docs/tutorials/intro-gpu.rst [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+4. Set the run parameters.
+
+   - We will use the name "Directions test 1"
+   - We will use the LMU-1 network to leverage the power of LMUs
```
Comment (drasmuss) [inline docs/tutorials/intro-gpu.rst:76]: LMU-1 -> LMU-large

File: docs/tutorials/intro-gpu.rst [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+Set up a full training run
+--------------------------
+
+Once you are satisfied with your test run,
+create a new run to train to convergence.
+
+1. Click on **Runs** in the top bar.
+
+2. Under the **Directions** heading, click **New run**.
```
Comment (drasmuss) [inline docs/tutorials/intro-gpu.rst:121]: +KWS

File: docs/tutorials/intro-gpu.rst [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+2. Download the :doc:`live microphone demo <../examples/microphone-demo/microphone-demo>`
+   to a local directory.
+
+3. Open NengoEdge to the main **Runs** page.
+
+4. Click the **Deploy** icon next to the fully trained run.
+
+5. Under the **Download model** heading,
+   click the **Format** dropdown and select **SavedModel**.
```
Comment (drasmuss) [inline docs/tutorials/intro-gpu.rst:175]: SavedModel is just called Binary now

File: docs/tutorials/uploading-datasets.rst [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+and ensure that all wav files can be loaded correctly.
+Feel free to use this script as a starting point for organizing your data!
+
+After running this script on the above data, we now have the directory structure that NengoEdge expects.
+
+.. image:: https://uploads-ssl.webflow.com/635051b5ae517f8542a06c6f/651711f2f6b267c58614fe72_Screenshot%20from%202023-09-29%2014-05-25.png
+
+We then create a compressed archive of this directory structure.
+Many tools will work for this,
+but we'll use the tar utility as it is usually available on Linux systems.
```
Comment (drasmuss) [inline docs/tutorials/uploading-datasets.rst:51]: We should use the `nengo-edge package-dataset` command here.

File: docs/tutorials/uploading-datasets.rst [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+To create a new dataset, click the **New dataset** button
+in the sections associated with your raw data on the **Datasets** page.
+
+With the example dataset from above,
+we created a dataset to classify the "aloha" keyword.
+All keywords not selected are grouped under the "unknown" label.
+Notice that the sample rate and clip duration are filled in according to the data,
+but can be changed if desired.
+Similarly, you can change the percentages associated with
+the amount of silence, unknown samples, and samples used for validation and testing.
```
Comment (drasmuss) [inline docs/tutorials/uploading-datasets.rst:94]: Validation/testing percentage are no longer options.

File: docs/glossary.rst [hunk 9]
```diff
@@ -0,0 +1,471 @@
+***************
+Important terms
+***************
```
Comment (drasmuss) [inline docs/glossary.rst:3]: After reading through these, I think I'm convinced that a list of terms is just not the best way to present this information. Removing the information from its context makes it all feel awkward. I think we can keep the content but move it all to a) tooltips, b) tours, or c) FAQs.

**Review by drasmuss — COMMENTED** [review]
I'll start on these fixups 👍

File: docs/faq.rst [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+representing background noise,
+which will be mixed in with the keyword audio during training.
+
+For example:
+
+* ``my_dataset``
+
+  * ``up``
+
+    * ``file0.wav``
```
[the remark this answers is not in the document]
Comment (tbekolay) [reply docs/faq.rst:26]: Yeah that makes sense; in the past I've used [`sphinx-tabs`](https://sphinx-tabs.readthedocs.io/en/latest/) for that, should be easy to use here too.

**Review by tbekolay — APPROVED** [review]
LGTM!

**Pull request merged**